types/styled-components/index.d.ts

```diff
@@ -88,10 +88,11 @@ type WithChildrenIfReactComponentClass<
 
 type StyledComponentPropsWithAs<
     C extends string | React.ComponentType<any>,
+    F extends string | React.ComponentType<any>,
```

Comment:
Introducing a new type parameter in the middle of existing type parameters is a breaking change, especially for anyone who might have an explicit `StyledCompontsPropsWithAs` reference in their code (a quick search of GitHub for just `StyledComponentPropsWithAs` shows this happens with some regularity).

Would it be feasible to add this as a trailing type parameter with a default `= C` to maintain backwards compatibility? If not, I'd need one of the definition owners to provide feedback before I'd feel comfortable approving this change.

Reply:
Yeah, that seems reasonable. It mangles the signatures a bit, but seems worth it for providing backwards compatibility.